packages/dev/core/src/Collisions/gpuPicker.ts

```diff
@@ -64,43 +65,73 @@ export class GPUPicker {
             const effect = this._renderMaterial!.getEffect();
 
             if (!mesh.hasInstances && !mesh.isAnInstance) {
-                effect.setColor4("color", this._idColors[mesh.uniqueId], 1);
+                effect.setColor4("meshID", this._idColors[mesh.uniqueId], 1);
             }
 
             this._meshRenderingCount++;
         };
 
-        this._renderMaterial.customBindingObservable.add(callback);
+        this._renderMaterial.onBindObservable.add(callback);
     }
 
     /**
      * Set the list of meshes to pick from
+     * Set that value to null to clear the list (and avoid leaks)
      * @param list defines the list of meshes to pick from
      */
     public setPickingList(list: Nullable<Array<AbstractMesh>>) {
-        if (!list) {
-            if (this._userDefinedList) {
-                for (let index = 0; index < this._pickableMeshes.length; index++) {
-                    const mesh = this._pickableMeshes[index];
-                    if (mesh.hasInstances) {
-                        (mesh as Mesh).removeVerticesData(VertexBuffer.ColorKind);
-                    }
+        if (this._pickableMeshes) {
+            // Cleanup
+            for (let index = 0; index < this._pickableMeshes.length; index++) {
+                const mesh = this._pickableMeshes[index];
+                if (mesh.hasInstances) {
+                    (mesh as Mesh).removeVerticesData(this._attributeName);
+                }
+                if (this._pickingTexure) {
+                    this._pickingTexure!.setMaterialForRendering(mesh, undefined);
                 }
             }
-            this._userDefinedList = false;
             this._pickableMeshes = [];
-            this._idMap = {};
+            this._idMap = [];
             this._idColors = [];

```

Comment:
nit pick - you can use `this._array.length = 0` to avoid GC of the array itself.